[Dispatch] Disable scatter fusion with producers

IanWood1 opened the pull request:
Backends don't currently support scatter fusion and will silently compile incorrect code. This should be turned off in order to prevent backends from generating incorrect results. I don't think any users are running into this currently, but its best to keep it off for now. Similar to https://github.com/iree-org/iree/pull/19535 but for both `indices` and `updates`.

MaheshRavishankar reviewed (commented):
Instead of this can we just disable this during dispatch region formation

IanWood1 committed:
[Dispatch] Disable scatter fusion with producers

Backends don't currently support scatter fusion and will silently
compile incorrect code. This should be turned off in order to prevent
backends from generating incorrect results.

Signed-off-by: Ian Wood <ianwood2024@u.northwestern.edu>

IanWood1 committed:
Disable fusion in dispatch formation

Signed-off-by: Ian Wood <ianwood2024@u.northwestern.edu>

hanhanW commented on compiler/src/iree/compiler/DispatchCreation/FormDispatchRegions.cpp:
```diff
@@ -651,7 +651,8 @@ isFusableWithProducer(OpOperand &operand,
 
   // Don't fuse attention with it's producer
-  if (isa<IREE::LinalgExt::AttentionOp>(consumer)) {
+  // TODO: Enable scatter fusion when supported by backends.
```
> Backends don't currently support scatter fusion and will silently compile incorrect code. 

Is there an issue for the TODO or your comment in the PR description?

hanhanW replied:
I found an issue about fusion (https://github.com/iree-org/iree/issues/19091), but not an issue for correctness. I mainly wonder if we should allocate someone from codegen side to fix the issue.

IanWood1 replied:
I think either @qedawkins or @MaheshRavishankar were going to look at this at some point. But I don't think there is a dedicated github issue for the correctness problem.
